Library/Homebrew/utils/lock.sh

```diff
@@ -38,14 +38,19 @@ _create_lock() {
   local lock_fd="$1"
   local name="$2"
   local ruby="/usr/bin/ruby"
+  local python="/usr/bin/python"
   [[ -x "$ruby" ]] || ruby="$(which ruby 2>/dev/null)"
+  [[ -x "$python" ]] || python="$(which python 2>/dev/null)"
 
-  if [[ -n "$ruby" ]]
+  if [[ -x "$ruby" && $("$ruby" -e "puts RUBY_VERSION >= '1.8.7' ? 0 : 1") = 0 ]]
```

Comment:
I'd be tempted to just make this something like:
```bash
if [[ -x "$ruby" && "$ruby" -e "exit(RUBY_VERSION >= '1.8.7')" ]]
```

That'll just use the exit code correctly.